pkg/prom/cleaner.go

```diff
@@ -0,0 +1,258 @@
+package prom
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/agent/pkg/prom/instance"
+	"github.com/grafana/agent/pkg/prom/wal"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	promwal "github.com/prometheus/prometheus/tsdb/wal"
+)
+
+const (
+	DefaultCleanupAge    = 12 * time.Hour
+	DefaultCleanupPeriod = 30 * time.Minute
+)
+
+var (
+	discoveryError = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "agent_prometheus_cleaner_storage_discovery_error",
+			Help: "Errors encountered discovering local storage paths",
+		},
+		[]string{"storage"},
+	)
+
+	segmentError = promauto.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "agent_prometheus_cleaner_segment_error",
+			Help: "Errors encountered finding most recent WAL segments",
+		},
+		[]string{"storage"},
+	)
```

Comment:
Can you rename the name of counters to end in `_total`? I'd like to stick to the defined Prometheus [best practices](https://prometheus.io/docs/practices/naming/) when possible.

Reply:
Sure, will do.